src/importexport/mei/internal/meiconverter.cpp

```diff
@@ -0,0 +1,1846 @@
+    case (engraving::AccidentalVal::FLAT3): return libmei::ACCIDENTAL_GESTURAL_tf;
+    case (engraving::AccidentalVal::SHARP): return libmei::ACCIDENTAL_GESTURAL_s;
+    case (engraving::AccidentalVal::SHARP2): return libmei::ACCIDENTAL_GESTURAL_ss;
+    case (engraving::AccidentalVal::SHARP3): return libmei::ACCIDENTAL_GESTURAL_ts;
+    default:
+        return libmei::ACCIDENTAL_GESTURAL_n;
+    }
+}
+
+engraving::BarLineType Convert::barlineFromMEI(const libmei::data_BARRENDITION meiBarline, bool& warning)
+{
+    warning = false;
+    switch (meiBarline) {
+    case (libmei::BARRENDITION_single): return engraving::BarLineType::NORMAL;
+    case (libmei::BARRENDITION_dbl): return engraving::BarLineType::DOUBLE;
+    case (libmei::BARRENDITION_rptstart): return engraving::BarLineType::START_REPEAT;
+    case (libmei::BARRENDITION_rptend): return engraving::BarLineType::END_REPEAT;
+    case (libmei::BARRENDITION_dashed): return engraving::BarLineType::BROKEN;
+    case (libmei::BARRENDITION_end): return engraving::BarLineType::END;
+    case (libmei::BARRENDITION_rptboth): return engraving::BarLineType::END_START_REPEAT;
+    case (libmei::BARRENDITION_dotted): return engraving::BarLineType::DOTTED;
+    case (libmei::BARRENDITION_heavy): return engraving::BarLineType::HEAVY;
+    case (libmei::BARRENDITION_dblheavy): return engraving::BarLineType::DOUBLE_HEAVY;
+    case (libmei::BARRENDITION_dbldashed):
+    case (libmei::BARRENDITION_dbldotted):
+        warning = true;
+        return engraving::BarLineType::DOUBLE;
+    case (libmei::BARRENDITION_dblsegno):
+        warning = true;
+        return engraving::BarLineType::DOUBLE;
+    case (libmei::BARRENDITION_segno):
+    case (libmei::BARRENDITION_invis):
+        warning = true;
+        return engraving::BarLineType::SINGLE;
+    default:
+        return engraving::BarLineType::SINGLE;
+    }
+}
+
+libmei::data_BARRENDITION Convert::barlineToMEI(const engraving::BarLineType barline)
+{
+    switch (barline) {
+    case (engraving::BarLineType::NORMAL): return libmei::BARRENDITION_NONE;
+    case (engraving::BarLineType::DOUBLE): return libmei::BARRENDITION_dbl;
+    case (engraving::BarLineType::START_REPEAT): return libmei::BARRENDITION_rptstart;
+    case (engraving::BarLineType::END_REPEAT): return libmei::BARRENDITION_rptend;
+    case (engraving::BarLineType::BROKEN): return libmei::BARRENDITION_dashed;
+    case (engraving::BarLineType::END): return libmei::BARRENDITION_end;
+    case (engraving::BarLineType::END_START_REPEAT):
+        return libmei::BARRENDITION_rptboth;
+    case (engraving::BarLineType::DOTTED): return libmei::BARRENDITION_dotted;
+    case (engraving::BarLineType::HEAVY): return libmei::BARRENDITION_heavy;
+    case (engraving::BarLineType::DOUBLE_HEAVY): return libmei::BARRENDITION_dblheavy;
+    case (engraving::BarLineType::REVERSE_END):
+        LOGD("Unsupported engraving::BarLineType::REVERSE_END");
```

Comment:
It doesn't matter, but we prefer to use the stream interface for the log (only in the old code there was a similar one)
```
LOGD() << "Unsupported engraving::BarLineType::REVERSE_END";
```